### PR TITLE
A few Spiro fixes

### PR DESCRIPTION
**Pull request by jtanx**
Related issues: #2390 

I think quite a few Spiro related crashes were the result of the fairly recent change made by a692c52.

* Fix crash/weird behaviour when deleting Spiro points (use after free issue; a692c52)
* Fix crash when deleting points and when the Spiro curve 'failed to converge' (e.g. where you see straight lines instead of Spiro curves)
* Fix the point info dialog (when you right-click on a control point and go to `Get Info...`) Ok/Cancel functions. (Before `Ok` did nothing for Spiro points, and I believe that `Cancel` didn't actually cancel/revert state for both Spiro and normal points)
* Don't display the Spiro end point when it 'fails to converge' e.g.
zeropt

cc @frank-trampe

### CODE REVIEW

File: fontforge/spiro.c [hunk 1]
```diff
@@ -231,7 +262,7 @@ void SSRegenerateFromSpiros(SplineSet *spl) {
 	SplinePoint *sp, *first, *last;
 	if ( (last=first=SplinePointCreate(spl->spiros[0].x,spl->spiros[0].y))==NULL )
 	    return;
-	for ( i=1; i<spl->spiro_cnt; ++i ) {
+	for ( i=1; spl->spiros[i].ty != SPIRO_END; ++i ) {
```
Comment (jtanx) [inline fontforge/spiro.c]: This addresses the last point I made in the PR comments

File: fontforge/splineutil.c [hunk 2]
```diff
@@ -1494,7 +1493,6 @@ static SplinePointList *SplinePointListSplitSpiros(SplineChar *sc,SplinePointLis
 	    memset(temp+(i-start),0,sizeof(spiro_cp));
 	    temp[i-start].ty = SPIRO_END;
 	    cur = SpiroCP2SplineSet( temp );
-	    free( temp ); temp = NULL;
```
Comment (frank-trampe) [inline fontforge/splineutil.c]: What keeps the reference to temp? Does it get freed somewhere else?

Comment (jtanx) [inline fontforge/splineutil.c]: `SpiroCP2SplineSet` sets `cur->spiros=temp` (i.e. `temp` is used in the return value)

**Comment by frank-trampe** [comment]
Looks good.

**Pull request merged**